### ISSUE TITLE
long running ofelia results in high memory usage

### ISSUE THREAD

**Issue by ticoombs** [mailcow/mailcow-dockerized#4272]
**Prior to placing the issue, please check following:** *(fill out each checkbox with an `X` once done)*
- [x] I understand that not following or deleting the below instructions will result in immediate closure and/or deletion of my issue.
- [x] I have understood that this bug report is dedicated for bugs, and not for support-related inquiries.
- [x] I have understood that answers are voluntary and community-driven, and not commercial support.
- [x] I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

## Summary

- ofelia-mailcow docker container seems to be running with a very high memory footprint, that only grows with time. 
- 5 hours of usage results in ~200MB
- Last updated 16/09/2021
- Possibly related to: https://github.com/mcuadros/ofelia/issues/157

Initial reboot usage:
`root      2817  0.3  3.5 251388 141872 ?       Ssl  07:31   0:00 /usr/bin/ofelia daemon --docker - TZ=Etc/UTC`
After 5 hours:
`root      8890  0.1  5.8 454484 231884 ?       Ssl  01:54   0:32 /usr/bin/ofelia daemon --docker - TZ=Etc/UTC`

Edit: 6 days of uptime:
`root      2817  0.1  5.0 522416 198476 ?       Ssl  Sep17  13:14 /usr/bin/ofelia daemon --docker - TZ=Etc/UTC`

## Logs

```
ofelia-mailcow_1     | 2021-09-17T07:28:37.004Z  common.go:125 ▶ NOTICE [Job "dovecot_imapsync_runner" (429894102126)] Started - /bin/bash -c "[[ ${MASTER} == y ]] && /usr/local/bin/gosu nobody /usr/local/bin/imapsync_runner.pl || exit 0"
ofelia-mailcow_1     | 2021-09-17T07:28:37.008Z  common.go:125 ▶ NOTICE [Job "sogo_ealarms" (887d79c42bbb)] Started - /bin/bash -c "[[ ${MASTER} == y ]] && /usr/local/bin/gosu sogo /usr/sbin/sogo-ealarms-notify -p /etc/sogo/sieve.creds || exit 0"
ofelia-mailcow_1     | 2021-09-17T07:28:37.016Z  common.go:125 ▶ NOTICE [Job "dovecot_trim_logs" (413f3d878d4c)] Started - /bin/bash -c "[[ ${MASTER} == y ]] && /usr/local/bin/gosu vmail /usr/local/bin/trim_logs.sh || exit 0"
ofelia-mailcow_1     | 2021-09-17T07:28:37.022Z  common.go:125 ▶ NOTICE [Job "sogo_sessions" (d9789abd4c07)] Started - /bin/bash -c "[[ ${MASTER} == y ]] && /usr/local/bin/gosu sogo /usr/sbin/sogo-tool expire-sessions ${SOGO_EXPIRE_SESSION} || exit 0"
ofelia-mailcow_1     | 2021-09-17T07:28:37.391Z  common.go:125 ▶ NOTICE [Job "dovecot_trim_logs" (413f3d878d4c)] Finished in "374.521728ms", failed: false, skipped: false, error: none
ofelia-mailcow_1     | 2021-09-17T07:28:37.521Z  common.go:125 ▶ NOTICE [Job "sogo_ealarms" (887d79c42bbb)] Finished in "512.414704ms", failed: false, skipped: false, error: none
ofelia-mailcow_1     | 2021-09-17T07:28:37.56Z  common.go:125 ▶ NOTICE [Job "dovecot_imapsync_runner" (429894102126)] Finished in "555.894881ms", failed: false, skipped: false, error: none
ofelia-mailcow_1     | 2021-09-17T07:28:37.645Z  common.go:125 ▶ NOTICE [Job "sogo_sessions" (d9789abd4c07)] Finished in "623.327562ms", failed: false, skipped: false, error: none
```

## Reproduction

Reproduction can be done by just having a default installation and waiting a couple hours. 

## System information

| Question | Answer |
| --- | --- |
| My operating system | Debian Buster (10) |
| Is Apparmor, SELinux or similar active? | No |
| Virtualization technlogy (KVM, VMware, Xen, etc - **LXC and OpenVZ are not supported** | KVM |
| Server/VM specifications (Memory, CPU Cores) | 2 Cores, 4GB Ram |
| Docker Version (`docker version`) | 19.03.13 |
| Docker-Compose Version (`docker-compose version`) | 1.25.5 |
| Reverse proxy (custom solution) | traefik using lables |

- Output of `git diff origin/master`, any other changes to the code? If so, **please post them**.
```
diff --git a/data/conf/rspamd/local.d/greylist.conf b/data/conf/rspamd/local.d/greylist.conf
index c43c907f..6b2ed8b8 100644
--- a/data/conf/rspamd/local.d/greylist.conf
+++ b/data/conf/rspamd/local.d/greylist.conf
@@ -2,3 +2,4 @@ whitelisted_ip = "http://nginx:8081/forwardinghosts.php";
 ipv4_mask = 24;
 ipv6_mask = 64;
 message = "Greylisted, please try again later";
+enabled = false;

```

- All third-party firewalls and custom iptables rules are unsupported. *Please check the Docker docs about how to use Docker with your own ruleset*. Nevertheless, iptabels output can help us to help you: `iptables -L -vn`, `ip6tables -L -vn`, `iptables -L -vn -t nat` and `ip6tables -L -vn -t nat`.
- DNS problems? Please run `docker exec -it $(docker ps -qf name=acme-mailcow) dig +short stackoverflow.com @172.22.1.254` (set the IP accordingly, if you changed the internal mailcow network) and post the output.

**Comment by mkuron** [comment]
I can confirm that, though it's not growing completely out of control. I have 514 MB of resident memory right now, after a 10-day uptime. That's obviously way too much for a simple Cron daemon, but still manageable with weekly restarts...

Is ofelia still being maintained? It hasn't received any commits in half a year and none of the maintainers have commented on any issues lately.

**Comment by dellarte** [comment]
Hi, I'm running mailcow (without clamav &  solr) on a 4GB server and I notice the same issue

**Comment by milkmaker** [comment]
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

**Comment by unixfox** [comment]
Bump still an issue

**Comment by milkmaker** [comment]
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

**Comment by unixfox** [comment]
Bump still an issue.

**Comment by ticoombs** [comment]
It looks like with the latest ofelia from Dec 31st memory usage has improved and is lower than before with ~5 days of uptime having ~150MB. 

`root     23129  0.3  3.8 1051600 152404 ?      Ssl  Jan31  18:43 /usr/bin/ofelia daemon --docker - TZ=Etc/UTC`

I think the main problem is the fact that mailcow _used_ to run on a 4GB system with memory to spare, but over time as clamav slowly took up more and more memory there was less and less memory to spare. Thus, when ofelia was chosen it was just the final item which caused our memory limited systems to have issues. The [Mailcow PreRequisites
](https://mailcow.github.io/mailcow-dockerized-docs/prerequisite-system/) page shows that 6GB of ram is "required" now, which is unfortunate but not without reason.

I think that _any_ extra docker containers that are added, will eventually be an issue regardless of the choices. Because of that I do not think #4362 would actually fix the underlying problem unless the memory footprint is **substantially** lower. (<50MB)

I am going to close this issue as my system has been stable for the past ~4 months.

Initial memory usage is now: 140M to 145M
`root     22565  0.5  3.6 848364 142976 ?       Ssl  22:25   0:01 /usr/bin/ofelia daemon --docker`

----

A note for anyone else, I have actioned the following items:

1. Disabled SOLR
 > Docker Variable: `mailcow__config_skip_solr: y`
2. Reduced SOGO web processes from 20 to 5
 > Modified Conf: `data/conf/sogo/sogo.conf:  WOWorkersCount = "5";`
3. Added an extra 2GB swap file on my system.
4. Regularly update mailcow
